azure/services/privateendpoints/privateendpoints.go

```diff
@@ -17,114 +17,34 @@ limitations under the License.
 package privateendpoints
 
 import (
-	"context"
-
-	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
+	asonetworkv1 "github.com/Azure/azure-service-operator/v2/api/network/v1api20220701"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
-	"sigs.k8s.io/cluster-api-provider-azure/azure/services/async"
-	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
-	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/aso"
 )
 
 // ServiceName is the name of this service.
 const ServiceName = "privateendpoints"
 
 // PrivateEndpointScope defines the scope interface for a private endpoint.
 type PrivateEndpointScope interface {
-	azure.Authorizer
-	azure.AsyncStatusUpdater
-	PrivateEndpointSpecs() []azure.ResourceSpecGetter
+	aso.Scope
+	PrivateEndpointSpecs() []azure.ASOResourceSpecGetter[*asonetworkv1.PrivateEndpoint]
 }
 
 // Service provides operations on Azure resources.
 type Service struct {
```

Comment:
I realized that these `Service` structs don't really seem necessary anymore. I removed them from my managedclusters PR: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4069/commits/2d37c52093d7537425a506a066ce9d1f971a274f

Reply:
Good find!

Reply:
Updated via 178d32f4039b9117767f49c599edcaf0bb1c202a